src/GraphQL.Tests/Language/CommentTests.cs

```diff
@@ -0,0 +1,203 @@
+using System.Linq;
+using GraphQL.Language;
+using GraphQL.Language.AST;
+using GraphQLParser;
+using Shouldly;
+using Xunit;
+
+namespace GraphQL.Tests.Language
+{
+    public class CommentTests
+    {
+        private readonly Parser _parser;
+
+        public CommentTests()
+        {
+            _parser = new Parser(new Lexer());
+        }
+
+        [Fact]
+        public void operation_comment_should_be_null()
+        {
+            const string query = @"
+query _ {
+    person {
+        name
+    }
+}";
+            
+            var document = CoreToVanillaConverter.Convert(query, _parser.Parse(new Source(query)));
+            document.Operations.First().Comment.ShouldBeNull();
+        }
+
+        [Fact]
+        public void operation_comment_should_not_be_null()
+        {
+            const string query = @"#comment
+query _ {
+    person {
+        name
+    }
+}";
+
+            var document = CoreToVanillaConverter.Convert(query, _parser.Parse(new Source(query)));
+            document.Operations.First().Comment.ShouldBe("comment");
+        }
+
+        [Fact]
+        public void query_comment_should_be_null()
+        {
+            const string query = @"
+query _ {
+    person {
+        name
+    }
+}";
+
+            var document = CoreToVanillaConverter.Convert(query, _parser.Parse(new Source(query)));
+            document.Operations.First().SelectionSet.Selections.OfType<Field>().First().Comment.ShouldBeNull();
+        }
+
+        [Fact]
+        public void query_comment_should_not_be_null()
+        {
+            const string query = @"
+query _ {
+    #comment
```

Comment:
It is not _query_ comment, it is field comment.

Reply:
ok